File: lib/taskgroups/ruby.rb
require_relative "../helpers"

class Ruby
  def self.l
    @l ||= SemanticLogger["misc"]
  end

  def self.rbenv
    if command? "rbenv"
      l.info "updating rbenv installation"
      (system "git", "-C", File.join(ENV["HOME"], ".rbenv"), "pull", "origin") || return
      self.gem
    else
      l.info "skipping; you don't use rbenv"
    end
  end

  def self.gem
    l.info "updating gems"
    system ( File.join ENV["HOME"], ".rbenv/shims/gem"), "update"
  end

  def self.all
    rbenv
  end
end

File: src/task/ruby.rs
use crate::{errors::Error, helpers::exec};
use log::info;
use std::{env, path::Path, process::Command};

pub fn update() -> Result<(), Error> {
    let home = env::var("HOME")?;
    let path = Path::new(&home).join(".rbenv");
    if path.is_dir() {
        info!("updating rbenv");
        exec(
            Command::new("git")
                .arg("-C")
                .arg(&path)
                .arg("pull")
                .arg("origin"),
        )?;
        exec(Command::new("gem").args(["update", "--system"]))?;
        exec(Command::new("gem").arg("update"))?;
    }
    Ok(())
}
